### PR TITLE
Datumises Flockminds

### PR DESCRIPTION
**Pull request by Mister-Moriarty**
[Gamemodes] [Internal] [Code Quality]

## About the PR:
Flockminds have been fully migrated onto the new datum system, including (future) roundstart, latejoining, random event, and admin-spawned flockminds. Flockmind abilities have also been separated out into distinct files.
Flocktraces will be migrated onto the new system at a later date, following the merge of subordinate antagonist datums introduced in #13346.

## Why's this needed?
Same rationale as #9366.

### CODE REVIEW

File: code/modules/antagonists/spy_thief/spy_thief.dm [hunk 1]
```diff
@@ -7,6 +7,15 @@
 	/// A list of items that this traitor has redeemed using their uplink. This tracks items redeemed with any uplink, so if a spy thief steals another spy thief's uplink, redeemed items will show up here too!
 	var/list/obj/redeemed_items = list()
 
+	New()
+		if (!ticker?.mode?.spy_market)
+			ticker.mode.spy_market = new /datum/game_mode/spy_theft
+			SPAWN(5 SECONDS)
```
Comment (ZeWaka) [inline code/modules/antagonists/spy_thief/spy_thief.dm:13]: ? missed?

File: code/mob/transform_procs.dm [hunk 2]
```diff
@@ -824,50 +824,15 @@ var/respawn_arena_enabled = 0
 			// no flocks given, make flockmind
 			message_admins("[key_name(usr)] made [key_name(src)] a flockmind ([src.real_name]).")
 			logTheThing(LOG_ADMIN, usr, "made [constructTarget(src,"admin")] a flockmind ([src.real_name]).")
-			return make_flockmind()
+			src.mind.add_antagonist(ROLE_FLOCKMIND)
+			return
```
Comment (ZeWaka) [inline code/mob/transform_procs.dm:828]: sanity check - this return? it's `null` which is different

File: code/modules/antagonists/spy_thief/spy_thief.dm (hunk 1 above)
Comment (Mister-Moriarty) [reply code/modules/antagonists/spy_thief/spy_thief.dm:13]: Unless I am misunderstanding, if either `ticker` or `mode` are null, the if statement will catch it; I don't see the necessity in checking twice.

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
This seems to have caused a runtime in the tutorial: 
image
which leads to the flock being stuck in psueudo-tutorial mode.
(yes `flock.flockmind_mind` is a var we actually need for... reasons)

File: code/datums/gamemodes/flock.dm [hunk 3]
```diff
@@ -28,13 +28,17 @@
...
 		possible_flockminds.Remove(flockmind)
 
 	return TRUE
 
+/datum/game_mode/flock/post_setup()
+	for (var/datum/mind/flockmind in flockminds)
+		flockmind.add_antagonist(ROLE_FLOCKMIND)
+
+	. = ..()
+
```
[the remark this answers is not in the document]
Comment (Mister-Moriarty) [reply code/datums/gamemodes/flock.dm:41]: `gm_parent` does not handle assigning antagonist roles at this time, although I do intend to refactor that at some point.

**Pull request merged**